node/Main.hs

```diff
@@ -68,7 +69,15 @@ action (SimpleNodeArgs (cArgs@CommonNodeArgs {..}) (nArgs@NodeArgs {..})) = do
     let vssSK = fromJust $ npUserSecret currentParams ^. usVss
     let gtParams = CLI.gtSscParams cArgs vssSK (npBehaviorConfig currentParams)
 
-    case sscAlgo of
+    let onNtpStatus = \case
+            NtpSyncOk -> logDebug $
+                      -- putText  $ -- FIXME: for some reason this message isn't printed
+                                    -- when using 'logDebug', but a simple 'putText' works
+                                    -- just fine.
+                "Local time is in sync with the NTP server"
+            NtpDesync diff -> logWarning $
+                "Local time is severely off sync with the NTP server: " <> show diff
+    withNtpCheck onNtpStatus $ case sscAlgo of
```

Comment:
It's better not to put any logic into `Main`, because it won't be shared between different executables.
We primarily need it for wallet (i. e. `cardano-node), not for core node (i. e. `cardano-node` simple). It also won't harm for explorer I suppose.
So I propose to have it in `Scehario.hs` probably.

Reply:
But we want to have different callbacks for `cardano-node` and `cardano-node-simple`. And we don't want to use `withNtpCheck` more than once.
So ok, I don't mind having `withNtpCheck` call in `Main`, but I think `onNtpStatus` should be defined somewhere in library (`infra` most likely). Maybe it will be used by explorer.